packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -401,7 +405,31 @@ export class NxMonorepoProject extends TypeScriptProject {
     this.updateWorkspace();
     this.wirePythonDependencies();
     this.synthesizeNonNodePackageJson();
+
+    // Prevent sub NodeProject packages from `postSynthesis` which will cause individual/extraneous installs.
+    // The workspace package install will handle all the sub NodeProject packages automatically.
+    const subProjectPackages: NodePackage[] = [];
+    this.subProjects.forEach((subProject) => {
+      if (isNodeProject(subProject)) {
+        const subNodeProject: NodeProject = subProject as NodeProject;
+        subProjectPackages.push(subNodeProject.package);
+        // @ts-ignore - `installDependencies` is private
+        subNodeProject.package.installDependencies = () => {};
```

Comment:
This is fantastic, although is there a way to achieve the same result using public API's?